src/test/java/org/jabref/rest/jabmap/resources/RootResourceTest.java

```diff
@@ -0,0 +1,479 @@
+import java.util.List;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.InternalField;
+import org.jabref.model.entry.field.MindMapField;
+import org.jabref.model.entry.types.MindMapEntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.model.jabmap.EdgeDirection;
+import org.jabref.model.jabmap.MindMapEdge;
+import org.jabref.model.jabmap.MindMapNode;
+import org.jabref.rest.jabmap.JabMapHTTPServer;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class RootResourceTest {
+
+    private static final String WEB_SERVICE_URI = "http://localhost:9898/";
+    private static Client client;
+
+    @BeforeAll
+    static void setUp() {
+        JabMapHTTPServer.startHttpEndPoint();
+        client = ClientBuilder.newClient();
+    }
+
+    @AfterAll
+    static void close() {
+        client.close();
+    }
+
+    @BeforeEach
+    void initMockDatabase() {
+        RootResource.databaseAccess = new MockDatabaseAccess();
+    }
+
+    @Test
+    void getReferenceEntries() {
+        List<BibEntry> bibEntries = setUpEntriesForGetReferenceEntries();
+        RootResource.databaseAccess.getActiveDatabase().insertEntries(bibEntries);
+
+        Response response = null;
+        try {
+            response = client.target(WEB_SERVICE_URI).path("libraries/current/entries").request().accept(MediaType.APPLICATION_JSON).get();
+            assertEquals(200, response.getStatus());
+            String responseBody = response.readEntity(String.class);
+            String expectedJson = "[{\"type\":{\"bibtex_metadata\":\"Article\",\"key\":\"article1\"}},{\"type\":{\"bibtex_metadata\":\"Book\",\"key\":\"book1\"}},{\"type\":{\"bibtex_metadata\":\"MastersThesis\",\"key\":\"MastersThesis1\"}}]";
+
+            assertEquals(expectedJson, responseBody);
+        } finally {
+            response.close();
+        }
+    }
+
+    @Test
+    void getMindMapEntries() {
+        List<BibEntry> bibEntries = setupExpectedForSaveCompleteMap();
+        RootResource.databaseAccess.getActiveDatabase().insertEntries(bibEntries);
+
+        Response response = null;
+        try {
+            response = client.target(WEB_SERVICE_URI).path("libraries/current/entries").request().accept(MediaType.APPLICATION_JSON).get();
+            assertEquals(200, response.getStatus());
+            String responseBody = response.readEntity(String.class);
+            String expectedJson = "[]";
+
+            assertEquals(expectedJson, responseBody);
+        } finally {
+            response.close();
```

Comment:
This reads like the try-with-resoures (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) should be used